### PR TITLE
feat(managedgroups): Classify resources for audit events

### PR DESCRIPTION
**Pull request by tmessi**
Includes some baseline tests, the necessary refactor and then
classification.

### CODE REVIEW

File: internal/proto/controller/api/resources/managedgroups/v1/managed_group.proto [hunk 1]
```diff
@@ -7,51 +7,62 @@ option go_package = "github.com/hashicorp/boundary/sdk/pbs/controller/api/resour
...
 
 // ManagedGroup contains all fields related to an ManagedGroup resource
 message ManagedGroup {
 	// Output only. The ID of the ManagedGroup.
-	string id = 10;
+	string id = 10; // @gotags: `class:"public"`
 
 	// Output only. Scope information for the ManagedGroup.
-	resources.scopes.v1.ScopeInfo scope = 20;
+	resources.scopes.v1.ScopeInfo scope = 20; // @gotags: `class:"public"`
```
Comment (johanbrandhorst) [inline internal/proto/controller/api/resources/managedgroups/v1/managed_group.proto:20]: I think we can remove this tag

File: internal/proto/controller/api/resources/managedgroups/v1/managed_group.proto [hunk 2]
```diff
@@ -7,51 +7,62 @@ option go_package = "github.com/hashicorp/boundary/sdk/pbs/controller/api/resour
...
+		google.protobuf.Struct attributes = 100 [
+      (custom_options.v1.generate_sdk_option) = true,
+      (custom_options.v1.subtype) = "default"
+    ];
+    OidcManagedGroupAttributes oidc_managed_group_attributes = 101 [
+			(google.api.field_visibility).restriction = "INTERNAL",
+			(custom_options.v1.generate_sdk_option) = true,
+			(custom_options.v1.subtype) = "oidc"
+		];
+  }
```
Comment (johanbrandhorst) [inline internal/proto/controller/api/resources/managedgroups/v1/managed_group.proto:55]: Mixing tabs and spaces here I think

File: internal/proto/controller/api/resources/managedgroups/v1/managed_group.proto [hunk 3]
```diff
@@ -7,51 +7,62 @@ option go_package = "github.com/hashicorp/boundary/sdk/pbs/controller/api/resour
...
-	// The attributes that are applicable for the specific ManagedGroup type.
-	google.protobuf.Struct attributes = 100 [(custom_options.v1.generate_sdk_option) = true];
+	string auth_method_id = 90 [json_name="auth_method_id", (custom_options.v1.subtype_source_id) = true]; // @gotags: `class:"public"`
+
+  oneof attrs {
+    // The attributes that are applicable for the specific ManagedGroup type.
+		google.protobuf.Struct attributes = 100 [
+      (custom_options.v1.generate_sdk_option) = true,
+      (custom_options.v1.subtype) = "default"
+    ];
```
Comment (johanbrandhorst) [inline internal/proto/controller/api/resources/managedgroups/v1/managed_group.proto:49]: I think we can classify this

File: internal/proto/controller/api/services/v1/managed_group_service.proto [hunk 4]
```diff
@@ -86,43 +89,43 @@ service ManagedGroupService {
...
-  string uri = 1;
+  string uri = 1; // @gotags: `class:"public"`
   resources.managedgroups.v1.ManagedGroup item = 2;
 }
 
 message UpdateManagedGroupRequest {
-  string id = 1;
+  string id = 1; // @gotags: `class:"public"`
   resources.managedgroups.v1.ManagedGroup item = 2;
   google.protobuf.FieldMask update_mask = 3 [json_name="update_mask"];
```
Comment (johanbrandhorst) [inline internal/proto/controller/api/services/v1/managed_group_service.proto:120]: I think we can classify this?

**Review by johanbrandhorst — APPROVED** [review]

**Review by johanbrandhorst — APPROVED** [review]

**Review by johanbrandhorst — APPROVED** [review]

**Review by jimlambrt — APPROVED** [review]

**Review by jefferai — APPROVED** [review]

**Pull request merged**